src/rpc/chain_api.rs

```diff
@@ -41,198 +42,91 @@ use tokio::sync::{
+    ChainGetParentMessages::register(module);
+    ChainGetMessage::register(module);
+    ChainGetParentReceipts::register(module);
+    ChainGetMessagesInTipset::register(module);
+    ChainExport::register(module);
+    ChainReadObj::register(module);
+    ChainHasObj::register(module);
+    ChainGetBlockMessages::register(module);
 }
 
-#[cfg(test)]
-quickcheck::quickcheck! {
-    fn quickcheck(val: PathChange) -> () {
-        assert_unchanged_via_json(val)
+pub enum ChainGetMessage {}
+impl RpcMethod<1> for ChainGetMessage {
+    const NAME: &'static str = "Filecoin.ChainGetMessage";
+    const PARAM_NAMES: [&'static str; 1] = ["msg_cid"];
+    const API_VERSION: ApiVersion = ApiVersion::V0;
+    type Params = (LotusJson<Cid>,);
+    type Ok = LotusJson<Message>;
+    async fn handle(
+        ctx: Ctx<impl Blockstore>,
+        (LotusJson(msg_cid),): Self::Params,
+    ) -> Result<Self::Ok, JsonRpcError> {
+        let chain_message: ChainMessage = ctx
+            .state_manager
+            .blockstore()
+            .get_cbor(&msg_cid)?
+            .with_context(|| format!("can't find message with cid {msg_cid}"))?;
+        Ok(LotusJson(match chain_message {
+            ChainMessage::Signed(m) => m.into_message(),
+            ChainMessage::Unsigned(m) => m,
+        }))
     }
 }
 
-pub async fn chain_get_message<DB: Blockstore>(
-    params: Params<'_>,
-    data: Ctx<DB>,
-) -> Result<LotusJson<Message>, JsonRpcError> {
-    let LotusJson((msg_cid,)): LotusJson<(Cid,)> = params.parse()?;
-
-    let chain_message: ChainMessage = data
-        .state_manager
-        .blockstore()
-        .get_cbor(&msg_cid)?
-        .with_context(|| format!("can't find message with cid {msg_cid}"))?;
-    Ok(LotusJson(match chain_message {
-        ChainMessage::Signed(m) => m.into_message(),
-        ChainMessage::Unsigned(m) => m,
-    }))
-}
-
-pub async fn chain_get_parent_messages<DB: Blockstore>(
-    params: Params<'_>,
-    data: Ctx<DB>,
-) -> Result<LotusJson<Vec<ApiMessage>>, JsonRpcError> {
-    let LotusJson((block_cid,)): LotusJson<(Cid,)> = params.parse()?;
-
-    let store = data.state_manager.blockstore();
-    let block_header: CachingBlockHeader = store
-        .get_cbor(&block_cid)?
-        .with_context(|| format!("can't find block header with cid {block_cid}"))?;
-    if block_header.epoch == 0 {
-        Ok(LotusJson(vec![]))
-    } else {
-        let parent_tipset = Tipset::load_required(store, &block_header.parents)?;
-        let messages = load_api_messages_from_tipset(store, &parent_tipset)?;
-        Ok(LotusJson(messages))
+pub enum ChainGetParentMessages {}
+impl RpcMethod<1> for ChainGetParentMessages {
+    const NAME: &'static str = "Filecoin.ChainGetParentMessages";
+    const PARAM_NAMES: [&'static str; 1] = ["block_cid"];
+    const API_VERSION: ApiVersion = ApiVersion::V0;
+    type Params = (LotusJson<Cid>,);
+    type Ok = LotusJson<Vec<ApiMessage>>;
+    async fn handle(
```

Comment:
Can we have a breather here and in other places to separate different logical pieces of the implementation?

```
    const NAME: &'static str = "Filecoin.ChainGetParentMessages";
    const PARAM_NAMES: [&'static str; 1] = ["block_cid"];
    const API_VERSION: ApiVersion = ApiVersion::V0;

    type Params = (LotusJson<Cid>,);
    type Ok = LotusJson<Vec<ApiMessage>>;

    async fn handle(
```

Reply:
While it's not a part of `rustfmt` (sadly), it's a standard found in most places, e.g., https://doc.rust-lang.org/std/str/trait.FromStr.html